Utils/adopt_utils.bash

```diff
@@ -0,0 +1,427 @@
+	else
+		date "+%B %d, %Y"
+	fi
+	
+}
+
+#######################################
+# Append adoption message to top of README.md 
+# Globals:
+#   None
+# Arguments:
+#   $1: the path to the Pack README.md
+#   $2: the message to write to the top of the README.md
+#######################################
+add_msg_to_readme(){
+
+	readme=$1
+	message=$2
+
+	if [ "$os" == "Mac OS" ] 
+	then
+		sed -i '' "1s/^/$message\n\n/g" "$readme"
+	else
+		sed -i "1s/^/$message\n\n/" "$readme"
+	fi
+
+}
+
+#######################################
+# Append adoption message to release note 
+# Globals:
+#   None
+# Arguments:
+#	$1: the option to add as release note
+#   $2: the path to the Pack release nope
+#######################################
+add_msg_to_rn(){
+	opt=$1
+	rn=$2
+
+	if [[ "$opt" == "complete" ]]; then
+		message="- Completed Adoption process."
+	else
+		message="- Started Adoption process."
+	fi
+
+	if [ "$os" == "Mac OS" ] 
+	then
+		sed -i '' "2s/.*/$message/" "$rn"
+	else
+		sed -i "2s/.*/$message/" "$rn"
+	fi
+
+}
+
+
+#######################################
+# Get Pack email
+# Globals:
+#   None
+# Arguments:
+#   $1: Pack path
+#######################################
+get_pack_email(){
+
+	pack_metadata="$1"
+	email=$(jq -r '.email' "$pack_metadata")
+
+	echo "$email"
+
+}
+
+#######################################
+# Get support link
+# Globals:
+#   None
+# Arguments:
+#   $1: Pack path
+#######################################
+get_pack_version(){
+
+	pack_metadata="$1"
+	version=$(jq '.currentVersion' "$pack_metadata")
+
+	echo "$version"
+
+}
+
+#######################################
+# Perform adoption steps mentioned in https://xsoar.pan.dev/docs/partners/adopt#process
+# 1) Create release notes and add message there
+# 2) Bump version
+# 3) Add message to README
+# Globals:
+#   None
+# Arguments:
+#   $1: Whether to start or complete adoption process
+#   $2: Path to Pack
+#   $3: Git branch that will hold all changes
+#######################################
+adopt() {
+	option=$1
+	dir=$2
+	branch=$3
+
+	readme="$dir/README.md"
+	pack_metadata="$dir/pack_metadata.json"
+	pack_name=$(basename "$dir")
+	
+
+	# Bump version and create release note file
+	demisto-sdk update-release-notes --input "$dir" --force --update-type "documentation" &> /dev/null
+	release_note=$(git --no-pager  diff --name-only --cached)
+	
+
+	pack_version=$(get_pack_version "$pack_metadata")
+	
+	echo "✓ Pack version bumped to $pack_version in '$pack_metadata'"
+	echo "✓ Release note created in '$release_note'"
+
+	# Add release note to second line (replacing documentation bullet)
+	add_msg_to_rn "$option" "$release_note"
+	release_note_name=$(basename "$release_note")
+	echo "✓ Release note '$release_note_name' updated."
+
+	# Add message to README
+	if [[ "$option" == "start" ]]; then
+		message="Note: Support for this Pack will be moved to Partner starting $(get_move_date)."
+	else
+		support_email=$(get_pack_email "$pack_metadata")
```

Comment:
This relies on the fact the partner has already added his email to the metadata... TBH since the adoption completion is more tricky we might need to think this through.

Reply:
Added a few methods to prompt and set email, URL and author image in [`d87017e` (#19115)](https://github.com/demisto/content/pull/19115/commits/d87017eb116998cb0520a8ccd9f0eb4bd0b8db64).